### PR TITLE
Refactor props to use ES6 spread

### PR DESCRIPTION
**Pull request by binoy14**
The advantage of doing this would be that all parent components can
be given native props for that component. This is also useful in case
any of react native props change/added/deleted we do not have to do
manual update of all components using that props.

### CODE REVIEW

**Comment by Monte9** [comment]
@binoy14 this is ready to be reviewed?

**Comment by binoy14** [comment]
@monte9 yes

**Comment by Monte9** [comment]
OK, will do. 

Couple general questions:
1. Does this PR cover all components? 
2. Also is there any TODO associated with this refactoring that we will have to fast-follow on?

**Comment by binoy14** [comment]
@monte9 

1. Yes, all components are covered
2. Not really, just need to test if it breaks anything or not

File: src/buttons/Button.js [hunk 1]
```diff
@@ -1,59 +1,53 @@
 import React, { PropTypes } from 'react';
-import { TouchableNativeFeedback, TouchableHighlight, StyleSheet, View, Platform, ActivityIndicator } from 'react-native';
+import {
+  TouchableNativeFeedback,
+  TouchableHighlight,
+  StyleSheet,
+  View,
+  Platform,
+  ActivityIndicator
+} from 'react-native';
```
Comment (Monte9) [inline src/buttons/Button.js]: I like this. Let's try stick to this formatting for multi-line imports.

File: src/buttons/Button.js [hunk 2]
```diff
@@ -1,59 +1,53 @@
...
-  color,
-  fontSize,
-  underlayColor,
-  raised,
-  textStyle,
-  large,
-  iconRight,
-  fontWeight,
-  disabledStyle,
-  fontFamily}) => {
```
Comment (Monte9) [inline src/buttons/Button.js]: Looking great!

File: src/buttons/Button.js [hunk 3]
```diff
@@ -95,24 +89,11 @@ const Button = ({
...
-      onPressIn={onPressIn}
-      onPressOut={onPressOut}
-      activeOpacity={activeOpacity}
-      onHideUnderlay={onHideUnderlay}
-      onShowUnderlay={onShowUnderlay}
-      background={background}
-      SelectableBackground={SelectableBackground}
-      SelectableBackgroundBorderless={SelectableBackgroundBorderless}
-      Ripple={Ripple}
-      hitSlop={hitSlop}
```
Comment (Monte9) [inline src/buttons/Button.js]: 👌

File: src/form/FormInput.js [hunk 4]
```diff
@@ -16,140 +15,36 @@ class FormInput extends Component {
...
+    return (
+      <View ref={containerRef} style={[styles.container, containerStyle && containerStyle]}>
+        <TextInput
+          ref={textInputRef}
+          selectionColor={selectionColor || colors.grey3}
+          style={[styles.input, inputStyle && inputStyle]}
+          {...attributes}
+        />
+      </View>
+    );
```
Comment (Monte9) [inline src/form/FormInput.js]: Hell yeah! This looks clean af! 💃

File: src/list/ListItem.js [hunk 5]
```diff
@@ -7,59 +7,61 @@ import colors from '../config/colors';
...
+    textInputMaxLength,
+    textInputMultiline,
+    textInputOnChangeText,
+    textInputOnFocus,
+    textInputOnBlur,
+    textInputSelectTextOnFocus,
+    textInputReturnKeyType,
+    textInputValue,
+    textInputStyle,
+    textInputContainerStyle,
```
Comment (Monte9) [inline src/list/ListItem.js]: can we get rid of these text input props in favor of `{...attributes}`?

**Comment by Monte9** [comment]
This looks great @binoy14! 

`+461 −666` That's a solid clean-up. Gj! 👏

**Pull request merged**